pkg/traceability/redaction/globalredaction.go

```diff
@@ -0,0 +1,130 @@
+package redaction
+
+import (
+	"regexp"
+
+	"github.com/Axway/agent-sdk/pkg/util/log"
+)
+
+// Global Agent redactions
+var agentRedactions Redactions
+
+//SetupGlobalRedaction - set up redactionRegex based on the redactionConfig
+func SetupGlobalRedaction(cfg Config) error {
+	var err error
+	agentRedactions, err = cfg.SetupRedactions()
+	return err
+}
+
+func setupShowRegex(showFilters []show) ([]showRegex, error) {
+	newShowRegex := make([]showRegex, 0)
+	for _, filter := range showFilters {
+		if filter.KeyMatch == "" {
+			continue // ignore blank keymatches as they match nothing
+		}
+		kc, err := regexp.Compile(filter.KeyMatch)
+		if err != nil {
+			log.Debug("Regex Error! Cannot Compile RequestHeader Remove Regex Keymatch", err)
+			return []showRegex{}, err
+		}
+
+		newShowRegex = append(newShowRegex, showRegex{
+			keyMatch: kc,
+		})
+	}
+	return newShowRegex, nil
```

Comment:
Can we put a debug before we leave showing the regex value

Reply:
Yes, I am actually going to add it to the error output, meant to handle that.